arclightning_frontend/games.html

```diff
@@ -27,41 +40,165 @@
+                    'meltyBlood3' :{name: "Melty Blood", description: "fighter with waifus", genre: ["zgenre", "anime", "2d"], thumbnail_path: 'images/meltyBloodWaifu.jpg'},
+                   'touhou3' : {name: "Touhou", description: "bullet hell with waifus", genre: ["bgenre", "anime"], thumbnail_path: 'images/touhouWaifu.png'},
+	}
+            },
+		
+	computed:{
+		alphSort: function() {
+			//idk if this even does anything
+			this.items = this.sortAlphabetical();
+			this.selectedIndex = Object.keys(this.items)[0];
+			return this.sortAlphabetical();
+		},
+		genreSort: function() {
+			this.items = this.sortGenre();
+			return this.sortGenre();
+		},
+		firstVal : function() {
+			this.selectedIndex = Object.keys(this.items)[0];
+			return Object.keys(this.items)[0];
+		}
+	
+	},
 
-                    {name: "Melty Blood", description: "fighter with waifus", genre: ["fighter", "anime", "2d"], thumbnail_path: 'images/meltyBloodWaifu.jpg', exe_path: 'C:\\Users\\THISUSER\\MELTYBLOOD_PATH'},
-                    {name: "Super Smash Bros", description: "crossover fighting game", genre: ["fighter", "crossover"], thumbnail_path: 'images/peach.png', exe_path: 'C:\\Users\\THISUSER\\SMASHBROS_PATH'},
-                    {name: "Melty Blood", description: "fighter with waifus", genre: ["fighter", "anime", "2d"], thumbnail_path: 'images/meltyBloodWaifu.jpg', exe_path: 'C:\\Users\\THISUSER\\MELTYBLOOD_PATH'},
-                    {name: "Touhou", description: "bullet hell with waifus", genre: ["bullet hell", "anime"], thumbnail_path: 'images/touhouWaifu.png', exe_path: 'C:\\Users\\THISUSER\\TOUHOU_PATH'},
+	methods: {
+		sendSelectedGame: function() {
+			//check to see if game exists
+			var gameSelected = this.items[this.selectedIndex];
+			if(gameSelected)
+				alert(this.selectedIndex)
+				axios.post('http://localhost:8000', {
+					id:this.selectedIndex
+			})
+		},
 
-	]
-            },
-        })
-        var keypress = new Vue({
-                el: '#keypress',
-                mounted: function () {
-                    document.addEventListener("keyup", (e) => {
-                        // redirect to demonstration.html if 'L' key is pressed
-                        if (e.keyCode == 76) { window.location.href = 'demonstration.html' }
-                    })
-                }
-            })
+		sortAlphabetical: function(){
+			//vue has a lot of great methods for arrays, however our game objects are dictionaries stored inside another dictionary. So my strategry here was to simply pipe the data into an array, use the sort method, then pipe it back into a dictionary, which dynamically updates site as keys are pressed
+			var sortedArr = [];
+			for(var key in this.items){
+				sortedArr[sortedArr.length] = key;
+			}	
+			sortedArr.sort();
+			var resultDict = {};
+			for(var i = 0; i < sortedArr.length; i++){
+				resultDict[sortedArr[i]] = this.items[sortedArr[i]];
+			}
+			this.items = resultDict;
+			return resultDict;			
```

Comment:
I don't think you have to return anything here since you are just trying to change the `items` dictionary which is done in the previous line. Same for the sortGenre() function.

Reply:
oh you are CORRECT !!! thank you